New RootQuerySelector (from 8.6.0) dissallows $expr at the query root (type error)

### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported


### Mongoose version

6.8.0

### Node.js version

20.15.0

### MongoDB server version

7.0.9

### Typescript version (if applicable)

5.5.4

### Description

Starting from mongoose 8.6.0, this raises a type error:

```ts
import type { FilterQuery } from "mongoose";

const filter: FilterQuery<{ owners: string[] }> = {
  $expr: { $lt: [{ $size: "$owners" }, 10] },
};
```

```text
Object literal may only specify known properties, and '$expr' does not exist in type 'FilterQuery<{ owners: string[]; }>'.ts(2353)
```

I think its related to the changes from #14764.

### Steps to Reproduce

Install mentioned versions and create a file with the snippet.

### Expected Behavior

No type errors, as its a [valid MongoDB query](https://www.mongodb.com/docs/manual/reference/operator/query/expr/).

I’m also having an issue with upgrading to 8.6 because it doesn’t recognize the field key in the filter of an updateMany query, which is part of the model.

` await this.updateMany({ url: { $in: urlsToCheck } }, { $set: { lastCheckAt: new Date() } });`

error TS2353: Object literal may only specify known properties, and 'url' does not exist in type 'RootFilterQuery<T>'.

--- a/test/types/queries.test.ts
+++ b/test/types/queries.test.ts
@@ -672,3 +672,9 @@ async function gh14545() {
   const myProjection = await M.findOne({}).select<SlimTest>({ prop: 1 }).exec();
   expectType<SlimTestDocument | null>(myProjection);
 }
+
+function gh14841() {
+  const filter: FilterQuery<{ owners: string[] }> = {
+    $expr: { $lt: [{ $size: "$owners" }, 10] },
+  };
+}

--- a/test/types/queries.test.ts
+++ b/test/types/queries.test.ts
@@ -675,6 +675,6 @@ async function gh14545() {
 
 function gh14841() {
   const filter: FilterQuery<{ owners: string[] }> = {
-    $expr: { $lt: [{ $size: "$owners" }, 10] },
+    $expr: { $lt: [{ $size: '$owners' }, 10] }
   };
 }

--- a/types/query.d.ts
+++ b/types/query.d.ts
@@ -116,6 +116,7 @@ declare module 'mongoose' {
     $where?: string | Function;
     /** @see https://www.mongodb.com/docs/manual/reference/operator/query/comment/#op._S_comment */
     $comment?: string;
+    $expr?: Record<string, any>;
     // we could not find a proper TypeScript generic to support nested queries e.g. 'user.friends.name'
     // this will mark all unrecognized properties as any (including nested queries) only if
     // they include a "." (to avoid generically allowing any unexpected keys)

